resources/assets/js/app/views/votes/ShowVote.js

```diff
@@ -46,16 +49,84 @@ module.exports = Marionette.LayoutView.extend({
 
         var self = this;
         this.collection.listenTo(Radio.channel('VoteComments'), 'newComment', function (comment) {
-                self.addedCommentsCollection.add(new CommentModel(comment), {parentUrl: ''});
-                var count = self.addedCommentsCollection.length + self.collection.length;
-                Radio.trigger('votesChannel', 'setCommentsCount' + self.options.voteModel.id, count);
-
-                if (comment.user_id != currentUser.id) {
-                    self.ui.newCommentButton.show(300);
-                } else {
-                    self.options.collection.add(self.addedCommentsCollection.toJSON());
-                }
-            });
+            self.addedCommentsCollection.add(new CommentModel(comment), {parentUrl: ''});
+            var count = self.addedCommentsCollection.length + self.collection.length;
+            Radio.trigger('votesChannel', 'setCommentsCount' + self.options.voteModel.id, count);
+
+            if (comment.user_id != currentUser.id) {
+                self.ui.newCommentButton.show(300);
+            } else {
+                self.options.collection.add(self.addedCommentsCollection.toJSON());
+            }
+        });
+
+
+        this.listenTo(Radio.channel('votesChannel'), 'saveUserChoice', function () {
+            self.ui.voteCommit.removeClass('disabled btn-default');
+            self.ui.voteCommit.addClass('btn-primary');
+        });
+    },
+
+    saveVotingOption: function () {
+        var self = this;
+
+        if (!self.ui.voteCommit.hasClass('disabled')) {
+            if (this.options.voteModel.get('is_single') == 1) {
+                var voteOption = this.$('input[name=optionsRadios]:checked').val();
+
+                var voteOptionModel = this.createVoteOptionModel(voteOption, 1);
+
+                voteOptionModel.save({}, {
+                    success: function () {
+                        self.$('i.voted').remove();
+                        self.ui.voteCommit.append(' <i class="glyphicon glyphicon-ok voted"></i>');
+                        self.ui.voteCommit.removeClass('btn-primary');
+                        self.ui.voteCommit.addClass('disabled btn-default');
+                    }
```

Comment:
Please move this part of code to a separate method to avoid code repeating. In Line 111